e2e/cli-e2e/tests/compare.e2e.test.ts

```diff
@@ -0,0 +1,58 @@
+import { simpleGit } from 'simple-git';
+import { ReportsDiff } from '@code-pushup/models';
+import { cleanTestFolder } from '@code-pushup/test-setup';
+import { executeProcess, readJsonFile } from '@code-pushup/utils';
+
+describe('CLI compare', () => {
+  const git = simpleGit();
+
+  beforeEach(async () => {
+    if (await git.diff(['--', 'examples/react-todos-app'])) {
+      throw new Error(
+        'Unstaged changes found in examples/react-todos-app, please stage or commit them to prevent E2E tests interfering',
+      );
+    }
+    await cleanTestFolder('tmp/e2e');
+    await executeProcess({
+      command: 'code-pushup',
+      args: ['collect', '--persist.filename=source-report'],
+      cwd: 'examples/react-todos-app',
+    });
+    await executeProcess({
+      command: 'npx',
+      args: ['eslint', '--fix', 'src', '--ext=js,jsx'],
+      cwd: 'examples/react-todos-app',
+    });
+    await executeProcess({
+      command: 'code-pushup',
+      args: ['collect', '--persist.filename=target-report'],
+      cwd: 'examples/react-todos-app',
+    });
+  }, 20_000);
+
+  afterEach(async () => {
+    await git.checkout(['--', 'examples/react-todos-app']);
+    await cleanTestFolder('tmp/e2e');
+  });
```

Comment:
Question. Shouldn't this be `afterAll`? The test (tests if there are any more in the future) cleans the folder in the `before` hook already so `afterEach` is questionable for me.

Reply:
There's only test, so I didn't think about this much. But using each hooks make more sense to be if there were more tests, each test should started from the clean state.